### PR TITLE
Use testng instead of junit

### PR DESCRIPTION
**Pull request by 315157973**
### Motivation
Pulsar uses TestNG, but some tests are still using junit

### Modifications
Use testng instead of junit

### CODE REVIEW

File: pulsar-functions/instance/src/test/java/org/apache/pulsar/functions/source/PulsarSourceTest.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
 import org.apache.pulsar.functions.api.Record;
 import org.apache.pulsar.functions.api.SerDe;
 import org.apache.pulsar.io.core.SourceContext;
-import org.junit.Assert;
+import org.testng.Assert;
```
[the remark this answers is not in the document]
Comment (315157973) [reply pulsar-functions/instance/src/test/java/org/apache/pulsar/functions/source/PulsarSourceTest.java:57]: Done

File: pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 import java.io.IOException;
 
-import static org.junit.Assert.assertEquals;
+import static org.testng.Assert.assertEquals;
```
Comment (hangc0276) [inline pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java:29]: Please take care of assertEquals order

**Review by hangc0276 — CHANGES_REQUESTED** [review]

File: pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java (hunk 2 above)
Comment (315157973) [reply pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java:29]: Already swapped

Comment (315157973) [reply pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java:29]: This was originally wrong, but it was correct after using testng

Comment (315157973) [reply pulsar-metadata/src/test/java/org/apache/pulsar/metadata/BacklogQuotaCompatibilityTest.java:29]: @hangc0276 PTAL

**Review by hangc0276 — APPROVED** [review]
LGTM

**Review by lhotari — APPROVED** [review]
LGTM

**Review by sijie — APPROVED** [review]

**Pull request merged**